### PR TITLE
fix(interactive): Support Index predicate for dynamic params in IR-Core

### PR DESCRIPTION
**Pull request by BingqingLyu**
## What do these changes do?

As titled.

## Related issue number

#3277

### CODE REVIEW

File: interactive_engine/executor/ir/proto/algebra.proto [hunk 1]
```diff
@@ -191,10 +191,16 @@ message Limit {
 // where the values referred by k1, k2, ... are indexed and hence the
 // predicate can be efficiently verified by leveraging the index.
 message IndexPredicate {
+  message PkValue {
```
Comment (longbinlai) [inline interactive_engine/executor/ir/proto/algebra.proto:194]: Looks like `PKValue` is not quite necessary here, Simply define 
```
message Triplet {
    common.Property key = 1;
    one of value {
         common.Value const = 2;
         common.DynamicParam param = 3;
    }
}
```

Comment (BingqingLyu) [reply interactive_engine/executor/ir/proto/algebra.proto]: Done

**Review by longbinlai — APPROVED** [review]
Can merge as long as cis passed

**Review by zhanglei1949 — APPROVED** [review]
LGTM

**Pull request merged**